src/backend/mmap.rs

```diff
@@ -0,0 +1,151 @@
+extern crate memmap;
+extern crate failure;
+
+use self::failure::ResultExt;
+
+use super::Backend;
+
+use ::error;
+use ::std::io;
+use ::std::cmp;
+
+#[derive(Debug)]
+struct Mmap {
+    inner: memmap::MmapMut,
+    //End of data
+    pub end: usize,
+    //Mmap total len
+    pub len: usize
+}
+
+impl Mmap {
+    fn new(len: usize) -> io::Result<Self> {
+        let inner = memmap::MmapOptions::new().len(len).map_anon()?;
+
+        Ok(Self {
+            inner,
+            end: 0,
+            len
+        })
+    }
+
+    fn as_slice(&self) -> &[u8] {
+        &self.inner[..self.end]
+    }
+
+    fn as_mut_slice(&mut self) -> &mut [u8] {
+        &mut self.inner[..self.end]
+    }
+
+    //Copies data to mmap and modifies data's end cursor.
+    fn write(&mut self, data: &[u8]) {
+        assert!(data.len() <= self.len);
```

Comment:
While in the current form this should never happen, please consider returning an error, something like: `You have encountered a bug in the Rustbreak memory mapping backend, please report it to your software's maintainer.`

Reason for that is that panicking in a library should only ever happen if there is an unrecoverable error. But in this case we can recover, by just ignoring the write and letting the application deal with this.